src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -476,26 +492,115 @@ private static DetailAST getLastLambdaToken(DetailAST lambda) {
     }
 
     /**
-     * Checks if switch member (case or default statement) is single-line statement, e.g.:
+     * Checks if current lambda belongs to a switch rule, e.g.:
      * <p>
      * {@code
-     * case 1: doSomeStuff(); break;
-     * case 2: doSomeStuff(); break;
+     * case 1 ->  monthString = "January";
+     * }
+     * </p>
+     *
+     * @param lambda {@link TokenTypes#LAMBDA}.
+     * @return true if current lambda belongs to a switch rule.
+     */
+    private static boolean isSwitchRuleLambda(DetailAST lambda) {
+        return lambda.getParent().getType() == TokenTypes.SWITCH_RULE;
+    }
+
+    /**
+     * Checks if current expression is a switch labeled expression . If so,
+     * braces are not allowed e.g.:
+     * <p>
+     * {@code
+     * case 1 -> 4;
+     * }
+     * </p>
+     *
+     * @param ast the ast to check
+     * @return true if current expression is a switch labeled expression.
+     */
+    private static boolean isSwitchLabeledExpression(DetailAST ast) {
+        final DetailAST parent = ast.getParent();
+        return switchRuleHasSingleExpression(parent);
+    }
+
+    /**
+     * Checks if current switch labeled expression contains only a single expression.
+     *
+     * @param switchRule {@link TokenTypes#SWITCH_RULE}.
+     * @return true if current switch rule has a single expression.
+     */
+    private static boolean switchRuleHasSingleExpression(DetailAST switchRule) {
+        final DetailAST possibleExpression = switchRule.findFirstToken(TokenTypes.EXPR);
+        return possibleExpression != null
+                && possibleExpression.getFirstChild().getFirstChild() == null;
+    }
+
+    /**
+     * Checks if switch member (case or default statement) in a switch rule or
+     * case group is on a single line.
+     *
+     * @param statement {@link TokenTypes#LITERAL_CASE case statement} or
+     * {@link TokenTypes#LITERAL_DEFAULT default statement}.
+     * @return true if current switch member is single-line statement.
+     */
+    private static boolean isSingleLineSwitchMember(DetailAST statement) {
+        final boolean isInSwitchRule =
+            statement.getParent().getType() == TokenTypes.SWITCH_RULE;
```

Comment:
This code duplicates method `isSwitchRuleLambda`. Should this method be used here?

Reply:
I've changed the method's name to  `isInSwitchRule()`, and I've used it in three more places.  Thanks for the recommendation.